ddp-workspace/projects/ddp-dsm-ui/src/app/file-download/file-download.component.ts

```diff
@@ -0,0 +1,66 @@
+import {DatePipe} from '@angular/common';
+import {Component, Input, Output} from '@angular/core';
+import {File} from '../participant-list/models/file.model';
+import {Participant} from '../participant-list/participant-list.model';
+import {ComponentService} from '../services/component.service';
+import {DSMService} from '../services/dsm.service';
+import {RoleService} from '../services/role.service';
+import {Utils} from '../utils/utils';
+
+const fileSaver = require( 'file-saver' );
+
+@Component( {
+  selector: 'app-file-download',
+  templateUrl: './file-download.component.html',
+  styleUrls: [ './file-download.component.scss' ]
+} )
+export class FileDownloadComponent {
+  @Input() participant: Participant;
+  downloadMessage = '';
+  downloading = false;
+
+  constructor( private role: RoleService, private dsmService: DSMService ) {
+
+  }
+
+  hasRole(): RoleService {
+    return this.role;
+  }
+
+  downloadParticipantFile( file: File ): void {
+    this.downloading = true;
+    if (file.scanResult !== 'CLEAN') {
+      this.downloadMessage = 'Error - file has not passed scanning';
+      this.downloading = false;
+      return;
+    }
+    const realm = localStorage.getItem( ComponentService.MENU_SELECTED_REALM );
+    this.dsmService.downloadParticipantFile( file.fileName, file.bucket, file.blobName, realm, file.mimeType ).subscribe(
```

Comment:
Can you please add `unsubscribe` for the subscription?

Reply:
Thanks! I wasn't aware we have to unsubscribe as well

Reply:
https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f